core/chains/evm/types/models.go

```diff
@@ -116,16 +118,27 @@ func (h *Head) IsInChain(blockHash common.Hash) bool {
 // HashAtHeight returns the hash of the block at the given height, if it is in the chain.
 // If not in chain, returns the zero hash
 func (h *Head) HashAtHeight(blockNum int64) common.Hash {
+	head := h.HeadAtHeight(blockNum)
+	if head == nil {
+		return common.Hash{}
+	}
+
+	return head.Hash
+}
+
+// HeadAtHeight returns the head at the given height, if it is in the chain.
+// If not in chain, returns nil
+func (h *Head) HeadAtHeight(blockNum int64) *Head {
```

Comment:
I'm not a big fan of this. We just added a new exported method that is only used by a single test. Can we get the top of the head on `head_saver_test.go` as it is and remove this?